jib-core/src/main/java/com/google/cloud/tools/jib/configuration/PortWithProtocol.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2018 Google LLC. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.configuration;
+
+/** Holds port number and protocol for an exposed port. */
+public class PortWithProtocol {
+
+  /** Represents the protocol portion of the port. */
+  public enum Protocol {
+    TCP,
+    UDP
+  }
+
+  private int port;
+  private Protocol protocol;
+
+  public PortWithProtocol(int port, Protocol protocol) {
+    this.port = port;
+    this.protocol = protocol;
+  }
+
+  public int getPort() {
+    return port;
+  }
+
+  public Protocol getProtocol() {
+    return protocol;
+  }
+
+  /** @return the {@link PortWithProtocol} in string form "port/protocol" */
+  @Override
+  public String toString() {
+    return port + "/" + protocol.toString().toLowerCase();
```

Comment:
Might be better to define the enum with the string form so that the enum name is decoupled from the string generation. But, it seems this string conversion should be handled in other places like when this is generated into the Dockerfile/container config and such.